doc/user/content/transform-data/strategies-considerations.md

```diff
@@ -0,0 +1,159 @@
+Consider rewriting your queries using idiomatic SQL.
+
+In addition, you can use various [query hints](/sql/select/#query-hints) with
+the idiomatic SQL to further improve the memory usage:
+
+| Query                           | Recommended Query Hint                  |
+|--------------------------------------|-----------------------------------------|
+| For First/Last value in group queries | `AGGREGATE INPUT GROUP SIZE`            |
+| For Top-K queries                    | `INPUT GROUP SIZE`                      |
+| For Top-1 queries using `DISTINCT ON` | `DISTINCT ON INPUT GROUP SIZE`          |
+
+### Index creation order
+
+An index can take advantage of the work performed by other indexes but only if
+these other indexes are created before it. That is, an index cannot take
+advantage of indexes that are created after it. However, during [blue-green
+deployments](/manage/dbt/development-workflows/#bluegreen-deployments) for your
+upgrades, the index creations ordering respects the dependency ordering.
+
+For example, if you create stacked views (i.e., views that depend on other
+views), we generally recommend that you start by creating an index only on the
+view that will serve results (unless otherwise suggested by the expected data
+access patterns). If later on, you also create an index on an intermediate view,
+the serving view index cannot benefit from the newly created index until the
+serving index is recreated after the new index, such as with a [blue-green
+[blue-green
+deployments](/manage/dbt/development-workflows/#bluegreen-deployments).
+
+## Other considerations
+
+Your workload may require considerations separate from the memory
+considerations.
+
+### Source envelope type
+
+The following considerations apply when using an upsert envelope:
+
+- Larger source cluster is required when using upsert envelope since in addition
+  to writing data to S3, a copy of each key consumed and the latest value is
+  maintained on the source cluster.
+
+- Using upsert envelope effectively dedupes your records (since a duplicated
+  update is effectively a no-op), but the initial hydration of sources takes
+  longer.
+
+### Materialized views and startup
+
+During startup, materialized views require memory proportional to both the input
+and output. When estimating required resources, consider both the startup cost
+and the steady-state cost.
+
+### Serve queries outside of Materialize
+
+Materialize can provide value by doing calculations on writes such that reads
+on these calculated values are fast.
+
+However, if you workload requires random access to a large dataset, it may be
+more feasible (because of memory and performance constraints) to serve queries
+from an external system instead of Materialize.  That is, have Materialize
+perform the various computations/transformations of the data and sink out to
+Kafka which then feeds into anothe system, from which you can perform your
+lookups.
+
+### Legacy sized clusters
```

Comment:
We may not want to list this -- but just in case, this comes up with enough frequency.

Reply:
Agree, I think it's better to have a direct discussion with customers on an as needed basis.